flict/impl.py

```diff
@@ -3,137 +3,108 @@
-                             "Missing argument to the verify command")
-
-    def _read_compliance_report(self, report_file):
-        on_error = "Internal error, report file has improper format, required: JSON"
-        try:
-            with open(report_file) as file_:
-                return json.load(file_)
-        except json.decoder.JSONDecodeError:
-            raise FlictError(ReturnCodes.RET_INTERNAL_ERROR, on_error)
-
-    def policy_report(self):
-        _compliance_report = self._read_compliance_report(self._args.report_file)
-        _policy = Policy(self._args.policy_file)
-        _policy_report = _policy.report(_compliance_report)
-        return self._formatter.format_policy_report(_policy_report)
+            with open(file_name) as fp:
+                return json.load(fp)[object_key]
 
-    def display_compatibility(self):
-        try:
-            # build up license string from all expressions
-            lic_str = " ".join(self._args.license_expression)
+    def _get_arbiter(self):
+        licenses_denied_file = self._args.licenses_denied_file
+        licenses_preference_file = self._args.licenses_preference_file
 
-            # encode (flict) all the license expression
-            lic_str = encode_license_expression(lic_str)
+        if self._args.licenses_info_file:
+            licenses_denied_file = self._args.licenses_info_file
+            licenses_preference_file = self._args.licenses_info_file
 
-            # build up license string from the expression string
-            _licenses = []
-            for lic in lic_str.split():
-                lic_list = self._license_handler.translate_and_relicense(lic).replace("(", "").replace(
-                    ")", "").replace(" ", "").replace("OR", " ").replace("AND", " ").strip().split(" ")
+        licenses_denied = self._read_json_object(licenses_denied_file, "licenses_denied", [])
+        licenses_preferences = self._read_json_object(licenses_preference_file, "license_preferences", [])
+        #parser = LicenseParserFactory.get_parser(licenses_preferences, licenses_denied)
+        arbiter = Arbiter(license_db=self._args.license_matrix_file, licenses_preferences=licenses_preferences, denied_licenses=licenses_denied)
 
-                for lic in lic_list:
-                    _licenses.append(decode_license_expression(lic))
+        return arbiter
 
-            # Diry trick to remove all duplicates
-            licenses = list(set(_licenses))
+    def verify(self):
+        arbiter = self._get_arbiter()
+        formatter = FormatterFactory.formatter(self._args.output_format)
 
-            compats = self._compatibility.check_compatibilities(licenses, self._args.extended_licenses)
-        except:
-            raise FlictError(ReturnCodes.RET_INVALID_EXPRESSSION,
-                             f'Could not parse license expression: {self._args.license_expression}')
+        if self._args.out_license and self._args.in_license_expr != []:
```

Comment:
` and any(self._args.in_license_expr)` or `and self._args.in_license_expr`

Reply:
You mean, replace:
`        if self._args.out_license and self._args.in_license_expr != []:`

with:
`        if self._args.out_license and any(self._args.in_license_expr) or and self._args.in_license_expr`

?

Reply:
yeah you got me confused with this `!= []` pattern - just replace that and it should be good

Reply:
Oh dear ... I missed that one of your "or" was not code :)